build/config/BUILDCONFIG.gn

```diff
@@ -15,6 +15,9 @@
 declare_args() {
   # Debug build. Set this to false to remove development features.
   is_debug = true
+
+  # Temporary flag for new interaction model engine, set it to true to enable
+  chip_app_use_interation_model = false
```

Comment:
This doesn't belong here; BUILDCONFIG.gn sets variables that are global to all build files including third party libraries. Not OK to add application or library specific options here. Instead, you can add it to a CHIP specific gni file and import it where it's needed.

Better yet, add this to CHIPConfig.h and just #include that, which is consistent with other config options.

Reply:
I think make it a config in gn will make it easier to test. And will will remove this flag when everything is ready.

Reply:
Moved this flag to src/app/common_flags.gni

Reply:
Note there are plenty of options in CHIPConfig.h that can be configured via GN (it generates CHIPBuildConfig.h, which is included into that config header).